### ISSUE TITLE
Multiple clients causing preset collapsing

### ISSUE THREAD

**Issue by kriselers** [antero111/plugin-presets#49]
I ran into an issue while playing on multiple clients using a preset I set up for my main and ironman account. It appears that the plugin currently saves a new copy of each config's .json each time a client is closed regardless of there being changes made to any of the presets or not. This of course creates an issue in a scenario like this:

* Make changes to plugin config A on client A
* Close client A, which causes all config files to save (Config A and B)
* Continue playing on client B
* Make changes to plugin config B on client B
* Close client B, which causes all config files to save overwriting any changes that were made to the plugin config A on client A since they weren't updated in client B

As for a potential solution, the first thing that immediately comes to mind is only saving the plugins when changes are made (i.e. the "update all" button is used) rather than the when the client closes. Secondly, you could maybe check if multiple clients are running and only allow changes to save from the first client opened or something along the lines of that. Granted, the first solution seems to be the most manageable in my opinion. I wish I could say that I had a proper suggestion on how this could be handled, but unfortunately I do not. This issue wasn't too big of a deal once I got the plugins to a state that there weren't many changes needed to be made. However, at some point all of my presets were deleted after closing my clients. I'm not too sure what would've caused this, so I can't point you in a particular direction on how to solve that. I would assume that it had something to do with having multiple clients open at one time.

**Comment by antero111** [comment]
Hi, thanks for the detailed writeup. I tried to follow the steps you described but haven't yet been able to reproduce this but only sometimes some presets disappear. You can restore them by clicking the "refresh presets" on the green + contex menu.

There is some edge cases where multiple clients cause preset collapsing but I haven't found any reliable way of testing them.  I'll have to test this more and get back to you if I find something. Otherwise your suggested fixes sound good.

To combat this, should there be a way for creating some sort of "preset list backup" e.g. all presets just stored in one file to store them outside of the ./presets folder? This would allow users to keep a backup of all presets if something like this arises again.

**Comment by kriselers** [comment]
Interesting that you weren't able to reproduce it as I was able to notice it pretty consistently. I'll create some backups of my current presets and see if I can come up with a more consistent process of reproducing it for you.

Thanks for the tip on the "refresh presets" option. I had no idea that was there. Funny enough, I believe that would've prevented me ever experiencing this issue for the most part except on those edge cases you mentioned. For example, refreshing the presets in client B once I made changes to preset A and closed client A. 

I do like the idea of having a way to backup your presets, especially since these files take up a very small amount of disk space (if the user if using local rather than cloud). However, I feel the important part of the backup is ensure that the backup is reliably protected from corruption/the multiple client save issue (which at this point, I'm only experiencing). This of course ties back to the suggestion of changing when the presets are saved and only saving the preset that has been modified.
